test/string-checker.js

```diff
@@ -207,6 +207,32 @@ describe('modules/string-checker', function() {
         });
     });
 
+    describe('esprima options', function() {
+        var customDescription = 'in no way a real error message';
+        var customEsprima = {
+            parse: function(code, options) {
+                var error = new Error();
+                error.description = options.foobar === 'qux' ? customDescription : 'should never happen';
+                error.lineNumber = 1;
+                error.column = 0;
+
+                throw error;
+            }
+        };
+
+        it('uses custom esprima options when set in the config', function() {
+            checker = new Checker({ esprima: customEsprima });
+            checker.registerDefaultRules();
+            checker.configure({ esprimaOptions: { foobar: 'qux' } });
+
+            var errors = checker.checkString('import { foo } from "bar";');
+            var error = errors.getErrorList()[0];
+
+            assert(error.rule === 'parseError');
+            assert(error.message === customDescription);
```

Comment:
Instead `assert(costumEsprima.parse.calledWith(expectedOptions))` which checks literally that the right options were called.

Once you have that set up, write a few more tests: 

1. one where you provide no options and see tolerant is used.
1. one where you provide options that aren't overwriteable.
1. one where you provide tolerant and see it's overriden.
1. one where you provide arbitrary options see that they're used.

P.S. see http://sinonjs.org/docs/#spies if you need more info on spies.